--- pkg-config-0.26/configure.ac.biarch	2011-05-15 14:31:43.000000000 +0200
+++ pkg-config-0.26/configure.ac	2011-05-21 12:10:35.000000000 +0200
@@ -30,6 +30,10 @@ fi
  AC_SUBST([pc_path])
 
 ])
+libdir_path32="$libdir/pkgconfig/32"
+AC_DEFINE_UNQUOTED(LIBDIR_PATH32, ["$libdir_path32"], [libdir pkgconfig path for 32 personality])
+pc_path32="$libdir/pkgconfig/32:$datadir/pkgconfig"
+AC_DEFINE_UNQUOTED(PKG_CONFIG_PC_PATH32, ["$pc_path32"], [path for 32 personality])
 
 PKG_CONFIG_FIND_PC_PATH
 
--- pkg-config-0.26/main.c.biarch	2011-05-15 11:06:06.000000000 +0200
+++ pkg-config-0.26/main.c	2011-05-21 12:16:34.000000000 +0200
@@ -29,6 +29,11 @@
 #include <string.h>
 #include <ctype.h>
 #include <stdio.h>
+#if defined(__linux__)
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/personality.h>
+#endif
 
 #ifdef G_OS_WIN32
 #define STRICT
@@ -171,6 +176,17 @@ print_hashtable_key (gpointer key,
   printf("%s\n", (gchar*)key);
 }
 
+#if defined(__linux__)
+static int
+is_dir(const char *path)
+{
+  struct stat st;
+  if (stat(path, &st) < 0)
+    return 0;
+  return S_ISDIR(st.st_mode);
+}
+#endif
+
 int
 main (int argc, char **argv)
 {
@@ -204,6 +220,9 @@ main (int argc, char **argv)
   char *search_path;
   char *pcbuilddir;
   gboolean need_newline;
+#if defined(__linux__)
+  long personality;
+#endif
   FILE *log = NULL;
   const char *pkgname;
 
@@ -304,6 +323,12 @@ main (int argc, char **argv)
     }
   else
     {
+#if defined(__linux__)
+      personality = syscall(SYS_personality, 0xffffffff);
+      if ((personality & PER_MASK) == PER_LINUX32 && is_dir(LIBDIR_PATH32))
+	      add_search_dirs(PKG_CONFIG_PC_PATH32, G_SEARCHPATH_SEPARATOR_S);
+      else
+#endif
       add_search_dirs(PKG_CONFIG_PC_PATH, G_SEARCHPATH_SEPARATOR_S);
     }
 
